src/piece/AbstractPiece.java

```diff
@@ -6,17 +6,19 @@
 	private Image image;
 	private Owner owner;
 	private int rank;
+	private Coordinate coordinate;
```

Comment:
This is refactored out later, we had thought that it was a good idea for pieces to know its location, but it turned out to be redundant info with the board keeping track of where pieces are.